### PR TITLE
Add IDs to headings and auto-link them

### LINKED ISSUE #57: Add anchors for section headers
It would be extremely useful to be able to have anchor IDs for each main section heading within a page so we can share URLs for discussion etc. Maybe show a link icon when hovering over the heading?

### PR DESCRIPTION
**Pull request by rowanmanning**
Now when you hover over certain headings on the component page, you get
a permalink to that heading. Resolves #57

registry-linked-headings

### CODE REVIEW

File: src/js/linked-heading.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+	/**
+	 * Build the heading link.
+	 */
+	buildLink() {
+		if (this.id) {
+			this.link = document.createElement('a');
+			this.link.setAttribute('href', `#${this.id}`);
+			this.link.setAttribute('title', 'Link directly to this section of the page');
+			this.link.innerHTML = '&nbsp;¶';
```
Comment (gvonkoss) [inline src/js/linked-heading.js]: Why is it a pilcrow?

Comment (notlee) [reply src/js/linked-heading.js]: So that's what that's called!

Comment (rowanmanning) [reply src/js/linked-heading.js]: Because that's what we use in techdocs :)

Comment (rowanmanning) [reply src/js/linked-heading.js]: We could change it if we have a better candidate. Though after a quick look, [Wikipedia says](https://en.wikipedia.org/wiki/Pilcrow):
> Online, it is used in some blogs and wikis to denote permalinks[13] (cf. Purple Numbers).

Comment (gvonkoss) [reply src/js/linked-heading.js]: Interesting... I had a very different understanding of the pilcrow. Here's some fun reading :D http://www.shadycharacters.co.uk/2011/02/the-pilcrow-part-1/ (@JakeChampion)

**Pull request merged**